Production_scripts/A05 Prepare Social Care Client lookup.R

```diff
@@ -0,0 +1,190 @@
+    alcohol = as.numeric(alcohol),
+    palliative_care = as.numeric(palliative_care),
+    carer = as.numeric(carer),
+    elderly_frail = as.numeric(elderly_frail),
+    neurological_condition = as.numeric(neurological_condition),
+    autism = as.numeric(autism),
+    other_vulnerable_groups = as.numeric(other_vulnerable_groups),
+    living_alone = as.numeric(living_alone),
+    support_from_unpaid_carer = as.numeric(support_from_unpaid_carer),
+    social_worker = as.numeric(social_worker),
+    type_of_housing = as.numeric(type_of_housing),
+    meals = as.numeric(meals),
+    day_care = as.numeric(day_care)
+  )
+
+
+# Data Cleaning ---------------------------------------
+
+## create outfile ##
+outfile <-
+  sc %>%
+  # sort
+  arrange(sending_location, social_care_id) %>%
+  # group
+  group_by(sending_location, social_care_id) %>%
+  # summarise to take last submission
+  summarise(
+    dementia = last(dementia),
+    mental_health_problems = last(mental_health_problems),
+    learning_disability = last(learning_disability),
+    physical_and_sensory_disability = last(physical_and_sensory_disability),
+    drugs = last(drugs),
+    alcohol = last(alcohol),
+    palliative_care = last(palliative_care),
+    carer = last(carer),
+    elderly_frail = last(elderly_frail),
+    neurological_condition = last(neurological_condition),
+    autism = last(autism),
+    other_vulnerable_groups = last(other_vulnerable_groups),
+    living_alone = last(living_alone),
+    support_from_unpaid_carer = last(support_from_unpaid_carer),
+    social_worker = last(social_worker),
+    type_of_housing = last(type_of_housing),
+    meals = last(meals),
+    day_care = last(day_care)
+  ) %>%
+  # recode missing with values
+  mutate(across(
+    .cols = c(
+      "support_from_unpaid_carer",
+      "social_worker",
+      "meals",
+      "living_alone",
+      "day_care"
+    ),
+    .x = replace_na(.x, 9)
+  )) %>%
+  mutate(
+    type_of_housing = replace_na(type_of_housing, 6)
+  ) %>%
+# factor labels
+  mutate(across(
+    c(
+      dementia,
+      mental_health_problems,
+      learning_disability,
+      physical_and_sensory_disability,
+      drugs,
+      alcohol,
+      palliative_care,
+      carer,
+      elderly_frail,
+      neurological_condition,
+      autism,
+      other_vulnerable_groups
+    ),
+    factor,
+    levels = c(0, 1),
+    labels = c("No", "Yes")
+  ),
+  across(
+    c(
+      living_alone,
+      support_from_unpaid_carer,
+      social_worker,
+      meals,
+      day_care
+    ),
+    factor,
+    levels = c(0, 1, 9),
+    labels = c("No", "Yes", "Not Known")
+  ),
+  type_of_housing = factor(type_of_housing,
+    levels = c(1:6),
+    labels = c(
+      "Mainstream", "Supported", "Long Stay Care Home",
+      "Hospital or other medical establishment", "Other",
+      "Not Known"
+    )
+  )
+  ) %>%
+# rename variables
+  rename_with(
+    .cols = -c(sending_location, social_care_id),
+    ~ paste0("sc_", .x)
+  )
```

Comment:
This is good and translates this fine from SPSS into R. However in the outfile it shows levels only and does not have 1 and 0 as flags. Need to figure out the best way to approach SPSS labels in R - TO DO: ask James opinon on this